server/sonar-server/src/main/java/org/sonar/server/search/Result.java

```diff
@@ -75,31 +72,52 @@ public Result(@Nullable BaseIndex<K, ?, ?> index, SearchResponse response) {
 
   private void processAggregation(Aggregation aggregation) {
```

Comment:
I'm not fond of creating helper classes that manipulate another class's internals. What could be done is to define a Facets class, embedded into Result<?>, which would be responsible for processing ES Aggregations and holding facet values.

WDYT ?